src/main/java/hudson/remoting/Channel.java

```diff
@@ -494,6 +519,8 @@ protected Channel(ChannelBuilder settings, CommandTransport transport) throws IO
 
         transport.setup(this, new CommandReceiver() {
             public void handle(Command cmd) {
+                commandsReceived++;
```

Comment:
Perhaps you meant reader reading incorrect value when write to normal non-volatile long variable can go in two 32bit writes ([JLS](http://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.7)). I've added volatile.